components/debounce/browser/debounce_rule.cc

```diff
@@ -187,17 +187,35 @@ bool DebounceRule::ValidateAndParsePatternRegex(
             << " which is an invalid regex pattern";
     return false;
   }
-  if (pattern_regex.NumberOfCapturingGroups() != 1) {
+  if (pattern_regex.NumberOfCapturingGroups() < 1) {
     VLOG(1) << "Debounce rule has param: " << pattern
-            << " which captures != 1 groups";
+            << " which captures < 1 groups";
     return false;
   }
 
-  if (!RE2::PartialMatch(path, pattern_regex, parsed_value)) {
+  // Get the number of capturing groups in the expression.
+  size_t number_of_capturing_groups =
+      pattern_regex.NumberOfCapturingGroups() + 1;
+
+  // Create result vectors to get the matches for the capturing groups.
+  std::vector<re2::StringPiece> match_results(number_of_capturing_groups);
+
+  if (!pattern_regex.Match(path, 0, path.size(), RE2::UNANCHORED,
+                           match_results.data(), match_results.size())) {
     VLOG(1) << "Debounce rule with param: " << param_
             << " was unable to capture string";
     return false;
   }
+
+  // Build parsed_value by appending matches, ignoring the first match which
+  // will be the whole match
+  std::for_each(std::begin(match_results) + 1, std::end(match_results),
```

Comment:
nit: I think you're already checking for a related failure, but I'd add a `DCHECK(match_results.size() > 1)` just to make it explicit that this line assumes there are at least two elements.